include/Synth/Synth.h

```diff
@@ -0,0 +1,118 @@
+
+#include "IMU/MPU6050.h"
+#include "Button/Button.h"
+#include <vector>
+#include <array>
+#include <string>
+#include <mutex>
+#include <atomic>
+#include <thread>
+#include <cmath>
+#include <iostream>
+#include <map>
+
+// STK includes
+#include <stk/Instrmnt.h>
+#include <stk/Rhodey.h>
+#include <stk/Wurley.h>
+#include <stk/Moog.h>
+#include <stk/Clarinet.h>
+#include <stk/RtAudio.h>
+#include <stk/Stk.h>
+
+// Define constants
+#define ACCEL_SCALE 16384.0
+#define G 9.81
+#define ACCEL_XOUT_H 0x3B
+#define ACCEL_YOUT_H 0x3D
+#define ACCEL_ZOUT_H 0x3F
+
+// Scale type definitions
+enum class ScaleType {
+    MAJOR,
+    MINOR,
+    PENTATONIC,
+    BLUES,
+    CHROMATIC
+};
+
+// Instrument type definitions
+enum class InstrumentType {
+    PIANO,
+    RHODEY,
+    WURLEY,
+    MOOG,
+    CLARINET
+};
+
+class MPUSynth {
+public:
+    MPUSynth(MPU6050* mpuSensor, int sampleRate = 44100);
+    ~MPUSynth();
+
+    // Start and stop audio processing
+    void start();
+    void stop();
+
+    // Set parameters
+    void setInstrument(InstrumentType instrumentType);
+    void setScale(ScaleType scaleType, const std::string& rootNote = "C");
+    void setVolume(float volume);
+    void setNoteDuration(float milliseconds);
+    
+    // Mappings
+    void setMappingMode(bool xAxis, bool yAxis, bool zAxis);
+    void setAxisNoteSpreads(int xSpread, int ySpread, int zSpread);
+    void setAxisVelocitySensitivity(float xSens, float ySens, float zSens);
+
+private:
+    // Note generation
+    void generateNotes(const std::string& rootNote, ScaleType scaleType);
+    int noteToMidiNumber(const std::string& note);
+    float midiNumberToFrequency(int midiNumber);
+    
+    // Audio callback and processing
+    static int audioCallback(void* outputBuffer, void* inputBuffer,
+                           unsigned int nBufferFrames,
+                           double streamTime, RtAudioStreamStatus status,
+                           void* userData);
+    void processSample(float* buffer, unsigned int nBufferFrames);
+    
+    // MPU6050 sensor data processing
+    void processMpuData();
+    void mapAccelerationToNotes(double x, double y, double z);
+    bool shouldTriggerNote(double acceleration, double prevAcceleration, double threshold);
+    
+    // STK and RtAudio related members
+    stk::Instrmnt* instrument;
+    RtAudio dac;
+    
+    // Audio parameters
+    int sampleRate;
+    float volume;
+    float noteDuration; // in milliseconds
+    std::atomic<bool> noteActive;
+    std::atomic<float> currentFrequency;
+    std::atomic<float> noteAmplitude;
+    std::atomic<float> noteTime; // current time within note duration
+    
+    // Note mapping parameters
+    std::vector<float> noteFrequencies;
+    std::array<bool, 3> axisEnabled; // x, y, z
+    std::array<int, 3> axisNoteSpread; // number of notes in range for each axis
+    std::array<float, 3> axisVelocitySensitivity;
+    
+    // Sensor related
+    MPU6050* mpuSensor;
```

Comment:
The MPU6050 object should not be directly used within the Synthesizer class. Instead, we need to register the Synthesizer object to the MPU6050's callbacks. See example in main for how we have done this for Button and MPU6050 (See line: https://github.com/Rapteon/air-synth/blob/dafedad0842541104758747f85b62ac3ed05c4c6/src/main.cpp#L24C1-L25C1)